### PR TITLE
Upgrade Airflow and dependencies

### PR DESCRIPTION
**Pull request by lwrubel**
Seems like a good time to get Airflow and dependencies updated and see what that process might involve for the future. 

I reviewed the Airflow suggested `docker-compose.yaml` for anything new with the latest release, but nothing more was involved there than updating the Airflow version number. 

To upgrade Python dependencies:

` uv pip compile pyproject.toml -o requirements.txt --upgrade`

Tested with unit tests and also running the DAG locally.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM apache/airflow:2.9.2-python3.12
+FROM apache/airflow:2.9.3-python3.12
```
Comment (edsu) [inline Dockerfile:1]: I was wondering if it would be living too dangerously to install the latest version?

**Review by edsu — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (lwrubel) [reply Dockerfile:1]: Upgrading Airflow feels like upgrading Rails to me. Do we upgrade patch version automatically there but not minor?

Comment (lwrubel) [reply Dockerfile:1]: I know some upgrades are a more manual process.

Comment (lwrubel) [reply Dockerfile:1]: I'll go ahead and merge this but open to figuring out how to loosen up the Airflow upgrade process.

**Pull request merged**